### ISSUE TITLE
Use the ISTIO_MULTIROOT_MESH to rotate root certificates

### ISSUE THREAD

**Issue by NikitaCloudRuntime** [istio/istio#50420]
Hello 

We would like to have Istio installed on multiple cluster  - multi-primary on multiple networks
At the same time we want to use shared root CA in order to build trust domain and use all Istio features between clusters
In order to set a shared CA we want to utilise Hashicorp Vault as root issuer and combine it with cert-manager deployed per cluster

In the described scenario, rotation of certificates for sidecar proxies is managed by the Istio control plane, rotation of certificate for istiod is managed by cert-manager but rotation of the root CA is still quite complex. Ito do this, we need to do an intermediate step when Istio proxies trust both the old and new CAs and then move everything to the second CA
Thisis described in details [here](https://blog.christianposta.com/diving-into-istio-1-6-certificate-rotation/). However, cert-manager creates `kubernetes.io/tls` secrets and does not have the ability to introduce public certificate of the second root issuer, hence we cannot use this method and cert-manager

What we wanted to do was to enable the `ISTIO_MULTIROOT_MESH`  feature and provide public certificate of the second root CA in yaml as shown in the following example
```
        meshConfig:
          defaultConfig:
            proxyMetadata:
              PROXY_CONFIG_XDS_AGENT: "true"
          caCertificates:
          - pem: |-
              -----BEGIN CERTIFICATE-----
              Root A
              -----END CERTIFICATE-----
              -----BEGIN CERTIFICATE-----
              Root B
              -----END CERTIFICATE-----
        components:
          pilot:
            k8s:
              env:
                - name: ISTIO_MULTIROOT_MESH
                  value: "true"
```
Thus, with that configuration, all sidecars would trust both existing root signed a current proxy certificate as well as the second root CA. After applying this configuration,  we can move to new root CA(second) and once all Istio proxies are updated disable `ISTIO_MULTIROOT_MESH`(and remove the above configuration)

Unfortunately, when we try to rotate certificates this way, once cert-manager creates new k8s secrets with the new certificate issued by the second root and Istiod picks it up, the following error message appears in the logs of Istiod
```
Updating a new ROOT-CA is not supported
```
This error comes from [here](https://github.com/istio/istio/blob/09c98bf9b125a21fc083da4d3606a71041b391a9/pilot/pkg/bootstrap/istio_ca.go#L330) and means that Istio is still expecting to see both roots in cacerts secret

So, we would like to ask for the ability to change this logic and enable `ISTIO_MULTIROOT_MESH` for root certificate rotation. This would make the process much easier

**Describe alternatives you've considered**

Possible alternatives are as follows
- doing the rotation manually
- doing the rotation with any IaC provisioner(terraform, crossplane, etc)

With both ways we will need to create `cacerts` secrets based on the output of the Hashicorp provider(to put both public certificates into `root-cert.pem` field) 

**Affected product area (please put an X in all that apply)**

[ ] Ambient
[ ] Docs
[ ] Dual Stack
[v ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[ v] Multi Cluster
[ ] Virtual Machine
[ v] Multi Control Plane

**Comment by volkorny** [comment]
@NikitaCloudRuntime We're experiencing the same issue in our mesh. Have you discovered any alternative solutions?

**Comment by Yufeireal** [comment]
+1 , facing the same issue.. Actively figuring out a 0-downtime solution which compatible with cert-manager tls cacerts secret

We host root ca in aws, use cert-manager to provision intermediate ca for istiod as intermediate ca;
Configuring` PROXY_CONFIG_XDS_AGENT: "true"` +  `caCertificates` + `ISTIO_MULTIROOT_MESH` seems to be working however, I do saw temporary TLS error and then recover itself.. But this means some downtime..

More curious about how PROXY_CONFIG_XDS_AGENT, caCertificates and ISTIO_MULTIROOT_MESH play the role here. 
I saw someone commented similar issue https://github.com/istio/istio/issues/39935

Ideally I think after I set these three params,
1.  Restart istiod and proxies, all workloads will trust both Root CA(plugin one and self-signed one).
2. Then I create cacerts(issued by the plugin root ca), no tls err expected?
But I do saw temporary TLS error 503... Which make me feel like these three params did not make proxy to trust both root? 
Also when I run this after step 1

` istioctl pc secret vauth-d7c97bdf8-25pfg -o json | jq '[.dynamicActiveSecrets[] | select(.name == "ROOTCA")][0].secret.validationContext.trustedCa.inlineBytes' -r | base64 -d | openssl x509 -noout -text`, still see the self-signed one as the root.. (maybe my command is not correct).
